blockauth/views/wallet_auth_views.py

```
@@ -130,3 +135,73 @@ def post(self, request):
+
+        except WalletConflictError:
+            self.link_throttle.record_failure(request, "wallet_link")
+            raise
+
+        except ValidationError as e:
+            self.link_throttle.record_failure(request, "wallet_link")
+            blockauth_logger.warning(
+                "Wallet link validation failed",
+                sanitize_log_context(request.data, {"errors": e.detail}),
+            )
+            raise ValidationErrorWithCode(detail=e.detail)
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)